### PR TITLE
feat(adr-036): Phase 1 skeleton (#848)

### PR DESCRIPTION
**Pull request by zjzcpj**
Closes #848

## Skeleton scope

This is the **Phase 1 skeleton** for ADR-036 (embedded code editor).
No real logic — every function body raises `NotImplementedError` (Python)
or `throw new Error(...)` (TypeScript) with a preceding docstring +
detailed implementation-plan comment + test plan + ADR reference.
Implementation phase agents (I36a / I36b / I36c) fill them in.

## Files added / modified

**Added:**
- `src/scieasy/api/routes/lint.py` — `POST /api/lint/python` stub
- `src/scieasy/blocks/_templates/__init__.py`
- `src/scieasy/blocks/_templates/block_base_template.py` — packaged "New custom block" scaffold
- `frontend/src/components/CodeEditor.tsx` — Monaco wrapper placeholder
- `tests/api/test_file_endpoints_skeleton.py` (11 xfail)
- `tests/api/test_lint_endpoint_skeleton.py` (6 xfail)
- `tests/api/test_blocks_template_skeleton.py` (3 xfail)
- `frontend/src/store/__tests__/tabState.skeleton.test.ts` (6 skip)
- `frontend/src/components/__tests__/CodeEditor.skeleton.test.tsx` (6 skip)

**Modified:**
- `src/scieasy/api/routes/projects.py` — file GET/PUT 501 stubs + Pydantic models + allowlist/size constants
- `src/scieasy/api/routes/blocks.py` — `GET /api/blocks/template` stub + response model
- `frontend/src/store/types.ts` — `WorkflowTab` + `FileTab` discriminated union; `TabState` aliased to `WorkflowTab` with **optional** `kind` for skeleton-phase backward compat (Phase 2A flips it to required + flips alias to true union)
- `frontend/src/store/tabSlice.ts` — `openFileTab` / `saveFileTab` / `updateFileTabContent` throw stubs
- `frontend/src/components/Toolbar.tsx` — `FileToolbar` shell + kind-switch scaffolding marked TODO; **all existing workflow buttons untouched** per dispatch scope
- `frontend/src/components/ProjectTree.tsx` — `handleDoubleClick` extended with TODO branches for `.py / .txt / .md / .json / .csv` outside `blocks/`
- `frontend/package.json` — lists `@monaco-editor/react ^4.6.0`
- `docs/planning/adr-035-036-checklist.md` — S36 rows ticked

## Comment-block coverage

Every `NotImplementedError` is preceded by a docstring containing:
Purpose, Signature contract, Implementation steps, Edge cases, Test plan,
References (ADR section + line numbers of mirrored existing code).
Implementation agents should be able to start coding without
re-reading the ADR.

## Test stubs

- 20 Python tests — all `@pytest.mark.xfail` with the test plan in the docstring
- 12 vitest tests — all `it.skip` with the test plan in the body comment

xfail (not skip) on the Python side surfaces accidental
implementation drift — if a body gets filled in without removing the
marker, vitest will flag it as XPASS.

## Checklist updates

Ticked rows in `docs/planning/adr-035-036-checklist.md` under
`Skeleton (S36)`:
- [x] all 11 rows under `### Skeleton (Owner: S36)` → branch `feat/issue-848/skeleton`

## Verification (locally before push)

- `ruff format --check .` → 461 files already formatted
- `ruff check .` → All checks passed
- `mypy` on changed Python files → Success
- `pytest --collect-only tests/api/` → 246 tests collected, no errors
- `pytest --no-cov tests/api/test_*_skeleton.py` → 20 xfailed (expected)
- `tsc -b && vite build` → built in 15.7s
- `vitest run` → 124 passed, 12 skipped (the new skeleton stubs)

## Dispatch deviation flagged

The dispatch prompt asked the template imports to literally be:
```python
from scieasy.blocks.base import Block, BlockSpec, PortSpec
```

These names are NOT exported from `scieasy.blocks.base`:
- `BlockSpec` lives in `scieasy.blocks.registry`
- The port classes are `InputPort` / `OutputPort` (no `PortSpec`)

The template uses the real exported names so the file actually compiles
(verified by importing `MyBlock`). This is documented in a top-of-file
comment in `block_base_template.py` for I36c to confirm/adjust during
implementation.

## NOTE for user

`frontend/package.json` now lists `@monaco-editor/react` as a dependency
but `npm install` was **not** run in this worktree (per the worktree
isolation rule — installs from a worktree do not propagate cleanly to
the user's main checkout and pollute global state). Please run in the
main checkout before Phase 2B work begins:

```
cd frontend && npm install
```

Until that runs, `npm run build` will fail in the main checkout because
`@monaco-editor/react` is unresolved. The skeleton component does NOT
import the package statically — Phase 2B (I36b) is responsible for the
lazy `import("@monaco-editor/react")` inside `useEffect`.

## CI status

[ ] All CI checks green (filled in after PR opens + checks complete)

### CODE REVIEW

File: src/scieasy/api/routes/blocks.py [hunk 1]
```diff
@@ -160,3 +161,59 @@ async def validate_connection_route(
...
+
+class BlockTemplateResponse(BaseModel):
+    """Skeleton — response shape for GET /api/blocks/template (per ADR-036 §3.12)."""
+
+    kind: str
+    content: str
+    suggested_filename: str
+
+
+@router.get("/template", response_model=BlockTemplateResponse)
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/blocks.py]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` order in `blocks.py` shows `GET /{block_type}/schema` and `GET /{block_type}` at lines 101/102, and the new `GET /template` at line 186. FastAPI routes `GET /api/blocks/template` to `get_block_schema` with `block_type="template"`, which calls `registry.get_spec("template")`, returns None → HTTP 404 "Unknown block type: template". The template endpoint is unreachable.

Fix: move the `@router.get("/template", ...)` decorator + handler above line 101 (`/{block_type}/schema`). F36-skeleton will reorder. The skeleton response still raises NotImplementedError, but at least I36c's implementation will be reachable once written.

File: frontend/package.json [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "@radix-ui/react-slot": "^1.2.4",
     "@radix-ui/react-tabs": "^1.1.13",
     "@radix-ui/react-tooltip": "^1.2.8",
+    "@monaco-editor/react": "^4.6.0",
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/package.json:19]: **accepted (will fix in audit-fix branch)** — verified P1.

Locally reproduced: with the committed `frontend/package-lock.json`, `npm ci` fails with:

```
npm error EUSAGE
npm error `npm ci` can only install packages when your package.json and package-lock.json are in sync.
npm error Missing: @monaco-editor/react@4.7.0 from lock file
npm error Missing: monaco-editor@0.55.1 from lock file
npm error Missing: @monaco-editor/loader@1.7.0 from lock file
npm error Missing: state-local@1.0.7 from lock file
npm error Missing: dompurify@3.2.7 from lock file
npm error Missing: marked@14.0.0 from lock file
npm error Missing: @types/trusted-types@2.0.7 from lock file
```

`.github/workflows/ci.yml` lines 151 & 188 both run `npm ci`, so this is a hard CI break the moment the umbrella PR runs frontend jobs. The skeleton agent's PR-body NOTE asked the user to run `npm install` in the main checkout, but the lockfile must travel WITH the PR — `npm ci` is the gating CI step.

Fix: F36-skeleton runs `npm install` in `frontend/`, commits the regenerated `package-lock.json` (~7 new packages including `@monaco-editor/react`, `monaco-editor`, `@monaco-editor/loader`, `state-local`, `dompurify`, `marked`, `@types/trusted-types`), pushes. Skeleton CodeEditor still does not statically import the package — Phase 2B will add the lazy import — so this only enables `npm ci` to succeed.

File: src/scieasy/api/routes/blocks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/blocks.py]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` order in `blocks.py` shows `GET /{block_type}/schema` and `GET /{block_type}` at lines 101/102, and the new `GET /template` at line 186. FastAPI routes `GET /api/blocks/template` to `get_block_schema` with `block_type="template"`, which calls `registry.get_spec("template")`, returns None → HTTP 404 "Unknown block type: template". The template endpoint is unreachable.

Fix: move the `@router.get("/template", ...)` decorator + handler above line 101 (`/{block_type}/schema`). F36-skeleton will reorder. The skeleton response still raises NotImplementedError, but at least I36c's implementation will be reachable once written.

[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/blocks.py]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` order in `blocks.py` shows `GET /{block_type}/schema` and `GET /{block_type}` at lines 101/102, and the new `GET /template` at line 186. FastAPI routes `GET /api/blocks/template` to `get_block_schema` with `block_type="template"`, which calls `registry.get_spec("template")`, returns None → HTTP 404 "Unknown block type: template". The template endpoint is unreachable.

Fix: move the `@router.get("/template", ...)` decorator + handler above line 101 (`/{block_type}/schema`). F36-skeleton will reorder. The skeleton response still raises NotImplementedError, but at least I36c's implementation will be reachable once written.

File: src/scieasy/api/routes/projects.py [hunk 3]
```diff
@@ -94,3 +95,157 @@ async def delete_project(project_id: str, runtime: RuntimeDep) -> None:
...
+
+
+class FileWriteResponse(BaseModel):
+    """Skeleton — response body shape for PUT file endpoint (per ADR-036 §3.2)."""
+
+    mtime: float
+    size: int
+
+
+@router.get("/{project_id:path}/file", response_model=FileReadResponse)
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/projects.py:102]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` declaration order in `projects.py` shows `/{project_id:path}` (GET/PUT) registered at lines 34/75/89, and the new `/{project_id:path}/file` handlers at 160/201. FastAPI matches first-declared first, so `GET /api/projects/<id>/file` is routed to `get_project` with `project_id="<id>/file"` — `runtime.open_project` then returns 404 (or attempts to open a path with a slash). The new file API is unreachable as committed.

Fix: move the two `/file` handlers above line 34 (or use a sub-router prefix). F36-skeleton will rebase the new endpoints to land BEFORE the greedy `{project_id:path}` group. Skeleton tests still xfail so the fix doesn't change test outcomes; runtime contract is preserved for I36a.

[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/projects.py:102]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` declaration order in `projects.py` shows `/{project_id:path}` (GET/PUT) registered at lines 34/75/89, and the new `/{project_id:path}/file` handlers at 160/201. FastAPI matches first-declared first, so `GET /api/projects/<id>/file` is routed to `get_project` with `project_id="<id>/file"` — `runtime.open_project` then returns 404 (or attempts to open a path with a slash). The new file API is unreachable as committed.

Fix: move the two `/file` handlers above line 34 (or use a sub-router prefix). F36-skeleton will rebase the new endpoints to land BEFORE the greedy `{project_id:path}` group. Skeleton tests still xfail so the fix doesn't change test outcomes; runtime contract is preserved for I36a.

File: frontend/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/package.json:19]: **accepted (will fix in audit-fix branch)** — verified P1.

Locally reproduced: with the committed `frontend/package-lock.json`, `npm ci` fails with:

```
npm error EUSAGE
npm error `npm ci` can only install packages when your package.json and package-lock.json are in sync.
npm error Missing: @monaco-editor/react@4.7.0 from lock file
npm error Missing: monaco-editor@0.55.1 from lock file
npm error Missing: @monaco-editor/loader@1.7.0 from lock file
npm error Missing: state-local@1.0.7 from lock file
npm error Missing: dompurify@3.2.7 from lock file
npm error Missing: marked@14.0.0 from lock file
npm error Missing: @types/trusted-types@2.0.7 from lock file
```

`.github/workflows/ci.yml` lines 151 & 188 both run `npm ci`, so this is a hard CI break the moment the umbrella PR runs frontend jobs. The skeleton agent's PR-body NOTE asked the user to run `npm install` in the main checkout, but the lockfile must travel WITH the PR — `npm ci` is the gating CI step.

Fix: F36-skeleton runs `npm install` in `frontend/`, commits the regenerated `package-lock.json` (~7 new packages including `@monaco-editor/react`, `monaco-editor`, `@monaco-editor/loader`, `state-local`, `dompurify`, `marked`, `@types/trusted-types`), pushes. Skeleton CodeEditor still does not statically import the package — Phase 2B will add the lazy import — so this only enables `npm ci` to succeed.

[the remark this answers is not in the document]
Comment (jiazhenz026) [reply frontend/package.json:19]: **accepted (will fix in audit-fix branch)** — verified P1.

Locally reproduced: with the committed `frontend/package-lock.json`, `npm ci` fails with:

```
npm error EUSAGE
npm error `npm ci` can only install packages when your package.json and package-lock.json are in sync.
npm error Missing: @monaco-editor/react@4.7.0 from lock file
npm error Missing: monaco-editor@0.55.1 from lock file
npm error Missing: @monaco-editor/loader@1.7.0 from lock file
npm error Missing: state-local@1.0.7 from lock file
npm error Missing: dompurify@3.2.7 from lock file
npm error Missing: marked@14.0.0 from lock file
npm error Missing: @types/trusted-types@2.0.7 from lock file
```

`.github/workflows/ci.yml` lines 151 & 188 both run `npm ci`, so this is a hard CI break the moment the umbrella PR runs frontend jobs. The skeleton agent's PR-body NOTE asked the user to run `npm install` in the main checkout, but the lockfile must travel WITH the PR — `npm ci` is the gating CI step.

Fix: F36-skeleton runs `npm install` in `frontend/`, commits the regenerated `package-lock.json` (~7 new packages including `@monaco-editor/react`, `monaco-editor`, `@monaco-editor/loader`, `state-local`, `dompurify`, `marked`, `@types/trusted-types`), pushes. Skeleton CodeEditor still does not statically import the package — Phase 2B will add the lazy import — so this only enables `npm ci` to succeed.

File: src/scieasy/api/routes/projects.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/api/routes/projects.py:102]: **accepted (will fix in audit-fix branch)** — verified P1.

`router.routes` declaration order in `projects.py` shows `/{project_id:path}` (GET/PUT) registered at lines 34/75/89, and the new `/{project_id:path}/file` handlers at 160/201. FastAPI matches first-declared first, so `GET /api/projects/<id>/file` is routed to `get_project` with `project_id="<id>/file"` — `runtime.open_project` then returns 404 (or attempts to open a path with a slash). The new file API is unreachable as committed.

Fix: move the two `/file` handlers above line 34 (or use a sub-router prefix). F36-skeleton will rebase the new endpoints to land BEFORE the greedy `{project_id:path}` group. Skeleton tests still xfail so the fix doesn't change test outcomes; runtime contract is preserved for I36a.
